web/source/osk/visualKeyboard.ts

```diff
@@ -480,20 +505,17 @@ namespace com.keyman.osk {
   }
 
   export class OSKSubKey extends OSKKey {
-    constructor(spec: OSKKeySpec, layer: string, formFactor: string) {
-      super(spec, layer, formFactor);
+    constructor(spec: OSKKeySpec, layer: string) {
+      if(typeof(layer) != 'string' || layer == '') {
+        throw "The 'layer' parameter for subkey construction must be properly defined.";
+      }
```

Comment:
What case is this covering off?

Reply:
From this PR's code changes:  https://github.com/keymanapp/keyman/blob/362344af165685f3d83d4e8102c88ad0b6184d80/web/source/osk/visualKeyboard.ts#L1768-L1772

Admittedly, the block above _should_ capture all cases that could trigger this scenario.  The other code path that contributes is seen here:  https://github.com/keymanapp/keyman/blob/362344af165685f3d83d4e8102c88ad0b6184d80/common/core/web/keyboard-processor/src/keyboards/activeLayout.ts#L144-L146

That said, noting the markedly different locations in play, I felt it wise to add a 'canary' in case a future change accidentally breaks either code path, as this particular aspect of the implementation isn't tightly encapsulated.  (Largely due to the difference between display logic & control paths for 'base' keys vs 'sub'keys.)